File: MergeSortedArray.swift
// Time Complexity : (O)m+n
// Space Complexity : (O)1
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this : 
// compare last element of array1(m-n) and array2 element. move the greatest number into last index of array 1
// if array2 is not completed then move all array2 element into array 1
class Solution {
    func merge(_ nums1: inout [Int], _ m: Int, _ nums2: [Int], _ n: Int) {
        var i = m-1
        var j = n-1
        var k = m+n-1
        
        while(i>=0 && j>=0) {
            if nums1[i] < nums2[j] {
                nums1[k] = nums2[j]
                j -= 1
            } else {
                nums1[k] = nums1[i]
                i -= 1
            }
            k -= 1
        }
        
        while j >= 0 {
            nums1[k] = nums2[j]
            j -= 1
            k -= 1
        }
        
    }
} 

File: twopointers2/src/mergesortedarray.java
class mergesortedarray {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        if(nums1==null || nums2==null){
            return;
        }
        int i=m-1; int j=n-1;
        int index=m+n-1;

        while(i>=0 && j>=0){
            if(nums1[i]>nums2[j]){
                nums1[index]=nums1[i];
                i-=1;
            }
            else{
                nums1[index]=nums2[j];
                j-=1;
            }
            index-=1;
        }

        while(j>=0){
            nums1[index]=nums2[j];
            j-=1;
            index-=1;
        }
    }
}

//time complexity=o(m+n)
//space complexity=o(1)